connect/api/src/main/java/org/apache/kafka/connect/connector/ConnectRecord.java

```diff
@@ -36,18 +36,19 @@
     private final Object key;
     private final Schema valueSchema;
     private final Object value;
+    private final Long timestamp;
 
-    public ConnectRecord(String topic, Integer kafkaPartition, Schema valueSchema, Object value) {
-        this(topic, kafkaPartition, null, null, valueSchema, value);
-    }
-
-    public ConnectRecord(String topic, Integer kafkaPartition, Schema keySchema, Object key, Schema valueSchema, Object value) {
```

Comment:
ConnectRecord is already abstract and it would be pretty weird for someone to extend it... Also, relocating it to another package would be a compatibility concern, no?